crates/ruff_python_ast/src/source_code/indexer.rs

```diff
@@ -114,4 +134,73 @@ import os
         let indexer: Indexer = lxr.as_slice().into();
         assert_eq!(indexer.continuation_lines(), [9, 12]);
     }
+
+    #[test]
+    fn string_lines() {
+        let contents = r#""this is a string""#;
+        let lxr: Vec<LexResult> = lexer::lex(contents, Mode::Module).collect();
+        let indexer: Indexer = lxr.as_slice().into();
+        assert!(
+            indexer.string_lines()[0].location.row() == 1
+                && indexer.string_lines()[0].end_location.row() == 1
+                && indexer.string_lines().len() == 1
+        );
+
+        let contents = r#"
+            """
+            this is a multiline string
+            """
+            "#;
+        let lxr: Vec<LexResult> = lexer::lex(contents, Mode::Module).collect();
+        let indexer: Indexer = lxr.as_slice().into();
+        assert!(
+            indexer.string_lines()[0].location.row() == 2
+                && indexer.string_lines()[0].end_location.row() == 4
+                && indexer.string_lines().len() == 1
+        );
+
+        let contents = r#"
+            """
+            '''this is a multiline string with multiple delimiter types'''
+            """
+            "#;
+        let lxr: Vec<LexResult> = lexer::lex(contents, Mode::Module).collect();
+        let indexer: Indexer = lxr.as_slice().into();
+        assert!(
+            indexer.string_lines()[0].location.row() == 2
+                && indexer.string_lines()[0].end_location.row() == 4
+                && indexer.string_lines().len() == 1
+        );
+
+        let contents = r#"
+            """this is a triple-quoted string on one line"""
+            "#;
+        let lxr: Vec<LexResult> = lexer::lex(contents, Mode::Module).collect();
+        let indexer: Indexer = lxr.as_slice().into();
+        assert!(
+            indexer.string_lines()[0].location.row() == 2
+                && indexer.string_lines()[0].end_location.row() == 2
+                && indexer.string_lines().len() == 1
+        );
+
+        let contents = r#"
+        """
```

Comment:
Could you add a test for a single line string?

Reply:
Yep, for sure. It also seems like there's a corner case I need to fix. With the current implementation, these two lines ([88](https://github.com/charliermarsh/ruff/blob/main/crates/ruff/resources/test/fixtures/pycodestyle/W19.py#L88), [137](https://github.com/charliermarsh/ruff/blob/main/crates/ruff/resources/test/fixtures/pycodestyle/W19.py#L137)) don't get counted as violations. I'll look into it after work, I'm wondering if it's maybe something where the tab character is on the line after a triple-quoted string.

Reply:
Actually, @MichaReiser I believe the single-line strings are covered on lines 140 and 176 of the file diff. Is that what you meant, or did you mean something like 
```
r""""multiline string on one line""""#;
```

Reply:
Though it also seems like `cargo test` doesn't run `indexer` unit tests. Is there a specific command we have for those?

Reply:
Uhm sorry. I meant adding a test for a single quote string (`'test'` or `"test"`) to make sure the indexer doesn't pick them up incorrectly.

Reply:
No problem! That's covered in the first test